rust/src/utils/kmeans.rs

```diff
@@ -334,6 +378,33 @@ impl KMeans {
         best_kmeans
     }
 
+    /// Create a kmeans model for a dataset that has less than `k` vectors.
+    ///
+    /// It assigns each vector as a centroid, and make the rest of the centroids
+    /// to be `f32::MAX`.
+    fn new_partial_kmeans(
```

Comment:
if we have 256 pq centroids, when would this be triggered in the test data we have?